youtube/views.py

```diff
@@ -8,14 +8,21 @@
 from .api import youtube_search
 
 YOUTUBE_RES_TYPES = ('video', 'channel', 'playlist')
-
+YOUTUBE_MAX_RESULTS = 50
+DEFAULT_MAX_RESULTS = 25
 
 def search(request):
     search_term = request.GET.get('search_term', '')
     restype = request.GET.get('restype', 'video')
-    max_results = request.GET.get('max_results', 25)
-    if max_results > 50:
-        max_results = 50
+    max_results = request.GET.get('max_results', DEFAULT_MAX_RESULTS)
+
+    try:
+        max_results = int(max_results)
+    except ValueError:
```

Comment:
`max_results` will not be None as an empty value for a query param will default to an empty string 
Eg:
```python
from django.http import QueryDict
GET = QueryDict('max_results')
GET.get('max_results')
```
In case `max_resuts` is missing it will default to  `DEFAULT_MAX_RESULTS`